doc/sphinx/practical-tools/utilities.rst

```diff
@@ -190,6 +198,57 @@ since it results in a “packed” plugin: All auxiliary modules (as
 (``Qux_plugin``). This reduces the chances of begin unable to load two
 distinct plugins because of a clash in their auxiliary module names.
 
+Comments
+++++++++
+Comments begin with a ``#`` at the start of a line, after whitespace or after a
+closing double quote. They continue to the end of the current line. The
```

Comment:
The part about "after a closing double quote" -- do you think that was a design intention or just a happenstance during coding?  If it's the latter we could consider whether to drop this feature in the interest of simplicity, which would have to be balanced against the expected impact on existing user code.

Reply:
I’d never have thought about this approach. Thanks for the input. I didn’t find any information about this, the code stems from https://github.com/coq/coq/commit/aad9936c225ef5b4f463b4af9ce1fa267bab337b.

The simplest behaviour for users would IMO be: "every `#` outside quotes starts a comment until the end of the current line". I don’t think a lot of users have strings with `#` in them, so I’ll do the cry-test and change the syntax accordingly.

We could mirror POSIX shell even more, by splitting `a "b"c` into `["a", "bc"]` instead of `["a", "b", "c"]` as coq_makefile does now. I think mirroring bash (or POSIX shell), is "very intuitive" behaviour for _CoqProject, since an unsuspecting user probably expects the same syntax in _CoqProject as in the shell.
This splitting is another reason why the passage you cited is so awkward.

Reply:
Handling `"b"c` that way seems odd.  I'm not sure why Posix did it that way or if it's really good to duplicate that behavior.  (Just my initial thought--maybe there was a good reason for Posix to do that.  And others may like it.  Note I won't be the reviewer for these code changes.)

Reply:
Hm. I’m fine with the current quoting, as long as it is documented.